easybuild/easyblocks/generic/cmakemakecp.py

```diff
@@ -0,0 +1,122 @@
+##
+# Copyright 2013 the Cyprus Institute
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: George Tsouloupas (The Cyprus Institute)
+@author: Fotis Georgatos (Uni.Lu, NTUA)
+@author: Kenneth Hoste (Ghent University)
+@author: Elizabeth Fischer (Columbia University)
+"""
+import os
+import shutil
+import glob
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import BUILD, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class CMakeMakeCp(CMakeMake):
+    """
+    CMake-based software with no make install step.
+    Running CMake is optional (see 'with_configure' in extra_options)
+    """
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """
+        Define list of files or directories to be copied after make
+        """
+        extra = {
+            'files_to_copy': [[], "List of files or dirs to copy", MANDATORY],
+            'with_configure': [False, "Run CMake before building", BUILD],
+        }
+        if extra_vars is None:
+            extra_vars = {}
+        extra.update(extra_vars)
+        return CMakeMake.extra_options(extra_vars=extra)
+
+    def configure_step(self, *args, **kwargs):
+        """
+        Configure build if required
+        """
+        if self.cfg.get('with_configure', False):
+            return super(CMakeMakeCp, self).configure_step(*args, **kwargs)
```

Comment:
if this is the only change from the `MakeCp` block, we should just inherit from both `MakeCp` and `CMakeMake` and only change the `configure_step`.